readthedocs/projects/views/public.py

```diff
@@ -363,10 +364,11 @@ def _version_same_domain_url(
             lang_slug=lang_slug,
             version_slug=version_slug,
         )
-        version = get_object_or_404(
-            final_project.versions.public(user=request.user),
-            slug=version_slug,
-        )
+
+        if not self.allowed_user(request, final_project, version_slug):
+            return self.get_unauthed_response(request, final_project)
+
+        version = final_project.versions.public(user=request.user).filter(slug=version_slug).first()
```

Comment:
Don't we need to raise a 404 here if `version == None`?

Reply:
True -- I will re-add that logic.